.github/workflows/doc.yml

```diff
@@ -18,13 +24,57 @@ jobs:
           profile: minimal
           override: true
 
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v1
+
       - name: Build docs
         env:
           RUSTDOCFLAGS: --cfg docsrs -D broken_intra_doc_links
         run: |
           cargo doc --workspace --no-deps --features=permission-calculator
           cargo doc -p twilight-util --no-deps --all-features
 
+      - name: Cache docs build
+        uses: actions/cache@v2
+        id: restore-docs
+        with:
+          path: ./*
+          key: ${{ github.sha }}
+
+  deploy-docs:
+    name: Deploy docs
+    runs-on: ubuntu-latest
+    needs: build-docs
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Download cached docs
+        uses: actions/cache@v2
+        id: restore-docs
+        with:
+          path: ./*
+          key: ${{ github.sha }}
+
+      - name: Remove examples
+        run: $(grep -h '^name' **/examples/**/Cargo.toml | cut -d'"' -f2 | sed 's/-/_/g' | xargs -I '{}' rm -rf ./target/doc/{})
```

Comment:
if you have a suggestion for something better i'd like to see it, i also find shell somewhat unreadable

Reply:
(This was also a pain to write so any improvement suggestions would be highly appreciated)

Reply:
`for example in $(grep -hoP '(?<=^name = ").+(?=")' **/examples/**/Cargo.toml); do rm -r target/doc/${example//-/_}; done` is an alternative but I don't know if it's much better (using bash builtin string manipulation instead of `sed` and regex instead of `cut`).
The fact that `Cargo.toml` needs to be read necessitates complexity

Reply:
Honestly there's probably a way to ls the target/doc directory and remove anything with examples from there, I don't know bash well enough to pull it off however

Reply:
It'd be very simple but the problem is that `twilight-lavalink-basic-lavalink-bot` dosen't contain `example`

Reply:
See #1243

Reply:
What should the new solution be once that gets merged?

Reply:
probably `rm -r target/doc/*_examples_*` (unless we skip building example docs with `cargo doc --no-deps --lib`)

Reply:
@7596ff?

Reply:
we should check as much as we can in the test, even if the examples don't have docs, then remove the directories we don't want to publish